packages/playwright-core/src/server/fetch.ts

```diff
@@ -172,20 +172,9 @@ export abstract class APIRequestContext extends SdkObject {
     // When `clientCertificates` is present, we set the `proxy` property to our own socks proxy
     // for the browser to use. However, we don't need it here, because we already respect
     // `clientCertificates` when fetching from Node.js.
-    if (proxy && !defaults.clientCertificates?.length && proxy.server !== 'per-context' && !shouldBypassProxy(requestUrl, proxy.bypass)) {
-      const proxyOpts = url.parse(proxy.server);
-      if (proxyOpts.protocol?.startsWith('socks')) {
-        agent = new SocksProxyAgent({
-          host: proxyOpts.hostname,
-          port: proxyOpts.port || undefined,
-        });
-      } else {
-        if (proxy.username)
-          proxyOpts.auth = `${proxy.username}:${proxy.password || ''}`;
-        // TODO: We should use HttpProxyAgent conditional on proxyOpts.protocol instead of always using CONNECT method.
-        agent = new HttpsProxyAgent(proxyOpts);
-      }
-    }
+    if (proxy && proxy.server !== 'per-context' && !shouldBypassProxy(requestUrl, proxy.bypass))
```

Comment:
I'd rather comment that `per-context` is here for historic reasons and backwards compatibility.